pkg/notifier/default.go

```diff
@@ -26,22 +25,27 @@ func (d *DefaultNotifier) Push(ctx context.Context, notification common.Notifica
 
 	recipientUser, err := d.userStore.Find(ctx, notification.Recipient())
 	if err != nil {
-		slog.Debug("failed to find user", "id", notification.Context().ID, "user", notification.Recipient().String(), "error", err)
-		return fmt.Errorf("failed to find recipient user: %w", err)
+		level := slog.LevelDebug
+		if !errors.Is(err, user.ErrUserNotFound) {
+			level = slog.LevelError
+		}
+		slog.Log(ctx, level, "failed to find user, will attempt to notify on provided identifiers", "id", notification.Context().ID, "user", notification.Recipient().String(), "error", err)
 	}
 
 	// The store may know of other identifiers for this user, so we merge those in
-	notification.Recipient().Merge(recipientUser.Identifiers)
+	if recipientUser != nil {
+		notification.Recipient().Merge(recipientUser.Identifiers)
+	}
 
 	for _, transport := range d.transports {
-		if !recipientUser.Wants(notification.Context().Type, transport.Key()) {
+		if recipientUser != nil && !recipientUser.Wants(notification.Context().Type, transport.Key()) {
 			slog.Debug("user does not want this notification this way", "id", notification.Context().ID, "user", recipientUser.String(), "transport", transport.Key())
 			continue
 		}
 
-		slog.Info("pushing notification", "id", notification.Context().ID, "type", notification.Context().Type, "user", recipientUser.String(), "transport", transport.Key())
+		slog.Info("pushing notification", "id", notification.Context().ID, "type", notification.Context().Type, "recipient", notification.Recipient().String(), "transport", transport.Key())
 		if err = transport.Push(ctx, notification); err != nil {
-			slog.Error("failed to push notification", "id", notification.Context().ID, "user", recipientUser, "transport", transport.Key(), "error", err)
+			slog.Error("failed to push notification", "id", notification.Context().ID, "recipient", notification.Recipient().String(), "transport", transport.Key(), "error", err)
```

Comment:
the downside here is we won't get a nice user key at the top. curious how folks would update this